Sources/StreamChatUI/Utils/MarkdownFormatter.swift

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public protocol MarkdownFormatter {
+    /// Checks for Markdown patterns in the given String.
+    /// - Parameter text: The string in which Markdown patters are going to be sought.
+    /// - Returns: Returns a Boolean value that indicates whether Markdown patters where found in the given String.
+    func containsMarkdown(text: String) -> Bool
+    /// Returns an attributed string form the given Markdown-formatted string.
+    /// - Parameter string: The string to be formatted.
+    /// - Returns: An attributed string with the corresponding formatted attributes.
+    func format(from string: String) -> NSAttributedString
+}
+
+/// Default implementation for the Markdown formatter
+class DefaultMarkdownFormatter: MarkdownFormatter {
+    enum Attributes {
+        enum Code {
+            static let fontName: String = "CourierNewPSMT"
+            static let color: UIColor = .red
+        }
+    }
+    
+    private let markdownRegex: String =
```

Comment:
is this that very that will never fail and where the force unwrap is acceptable?
```suggestion
    private let markdownRegex = NSRegularExpression(
      pattern: "((?:\\`(.*?)\\`)|(?:\\*{1,2}(.*?)\\*{1,2})|(?:\\~{2}(.*?)\\~{2})|(?:\\_{1,2}(.*?)\\_{1,2})|^(>){1}|(#){1,6}|(=){3,10}|(-){1,3}|(\\d{1,3}\\.)|(?:\\[(.*?)\\])(?:\\((.*?)\\))|(?:\\[(.*?)\\])(?:\\[(.*?)\\])|(\\]\\:))+", 
      options: .anchorsMatchLines
     )!
```

Reply:
Not the case though. As you can see in the guard statement in containsMarkdown() the NSRegularExpression  init throws because it indeed can fail so is easier to treat it as an optional with `try?` .